src/devices/Gpio/Drivers/Sunxi/SunxiDriver.cs

```diff
@@ -0,0 +1,469 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Device.Gpio.Drivers;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Iot.Device.Gpio.Drivers
+{
+    /// <summary>
+    /// A generic GPIO driver for Allwinner SoCs.
+    /// </summary>
+    /// <remarks>
+    /// This is a generic GPIO driver for Allwinner SoCs.
+    /// It can even drive the internal pins that are not drawn out.
+    /// Before you operate, you must be clear about what you are doing.
+    /// </remarks>
+    public unsafe partial class SunxiDriver : GpioDriver
+    {
+        private IntPtr _gpioPointer0;
+        private IntPtr _gpioPointer1;
+
+        /// <summary>
+        /// CPUX-PORT base address. 
+        /// </summary>
+        protected virtual int CpuxPortBaseAddess { get; }
+        /// <summary>
+        /// CPUS-PORT base address.
+        /// </summary>
+        protected virtual int CpusPortBaseAddess { get; }
+        // final_address = mapped_address + (target_address & map_mask) https://stackoverflow.com/a/37922968
+        private readonly int _mapMask = Environment.SystemPageSize - 1;
+
+        private static readonly object s_initializationLock = new object();
+        private static readonly object s_sysFsInitializationLock = new object();
+        private const string GpioMemoryFilePath = "/dev/mem";
+        private UnixDriver _interruptDriver;
+        private readonly IDictionary<int, PinState> _pinModes = new Dictionary<int, PinState>();
+
+        protected SunxiDriver() { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SunxiDriver"/>.
+        /// </summary>
+        /// <param name="cpuxPortBaseAddess">CPUX-PORT base address (This can be find in the corresponding SoC datasheet).</param>
+        /// <param name="cpusPortBaseAddess">CPUS-PORT base address (This can be find in the corresponding SoC datasheet).</param>
+        public SunxiDriver(int cpuxPortBaseAddess, int cpusPortBaseAddess)
+        {
+            CpuxPortBaseAddess = cpuxPortBaseAddess;
+            CpusPortBaseAddess = cpusPortBaseAddess;
+        }
+
+        /// <summary>
+        /// The number of pins provided by the driver.
+        /// </summary>
+        protected internal override int PinCount => throw new PlatformNotSupportedException("This driver is generic so it can not enumerate how many pins are available.");
```

Comment:
Once you change to start referencing System.Device.Gpio as opposed to compilingt he source as you are doing now, you will have to change all of these abstract method signatures to instead be `protected override ...`. So dropping the internal given that via a reference you won't be seeing that any longer.